### ISSUE TITLE
[Bug]:[Memory] memory leak on 2.0.0-rc8 

### ISSUE THREAD

**Issue by rchangj** [milvus-io/milvus#12423]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.0.0-rc8 
- Deployment mode(standalone or cluster):standalone
- SDK version(e.g. pymilvus v2.0.0rc2):2.0.0rc8
- OS(Ubuntu or CentOS): Ubuntu
- CPU/Memory: 128G
- GPU: N/A
- Others:
```

### Current Behavior

When I ran Milvus 2.0.0-rc8 in standalone mode on ubuntu 20.2. I noticed followings:
1.Memory leak, both with/without loading the collection, there is about 3-4 GB memory lost. The lost memory seems to be added to buff/cache. Following is a snapshot after several days run. The system has been idle during the time.
MiB Mem : 127350.3 total,  98087.9 free,   2992.0 used,  26270.4 buff/cache

Can someone help take a look at this issue? Thanks.

### Expected Behavior

No outstanding memory leak.

### Steps To Reproduce

```markdown
1. installation steps:
https://milvus.io/docs/install_standalone-docker.md

2. load vector, about 2k

3.system run idle and use top to check the memory usage
```

**Comment by yanliang567** [comment]
@scsven  could you please take a look at this issue? I think this is a known issue to you.
/assign @scsven 
/unassign

**Comment by rchangj** [comment]
Hi, @yanliang567 and @scsven 
I have another question which might be related to this issue. But if you believe it's a separate issue, I can open a separate bug. 
Under same setup, even when system is idle, I kept seeing sst files generated. Averagely several GB/day. Here is a snapshot of the log from rib_data:

** Compaction Stats [default] **
Level    Files   Size     Score Read(GB)  Rn(GB) Rnp1(GB) Write(GB) Wnew(GB) Moved(GB) W-Amp Rd(MB/s) Wr(MB/s) Comp(sec) CompMergeCPU(sec) Comp(cnt) Avg(sec) KeyIn KeyDrop
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  L0      1/0   35.16 MB   0.2      0.0     0.0      0.0      15.3     15.3       0.0   1.0      0.0    260.7     60.02             38.09       445    0.135       0      0
  L1      4/0   212.94 MB   0.8     39.1    15.2     23.9      39.1     15.2       0.0   2.6    184.9    184.9    216.71            171.32       111    1.952    235M    50K
  L2     48/0    2.39 GB   1.0     49.2    15.0     34.3      48.1     13.8       0.3   3.2    177.3    173.2    284.25            234.43       224    1.269    355M    17M
  L3    398/0   24.95 GB   1.0     15.8    13.9      1.9      15.8     13.9       0.3   1.1    241.6    241.6     67.03             35.35        60    1.117     32M    101
  L4     78/0    4.88 GB   0.0      0.0     0.0      0.0       0.0      0.0       4.9   0.0      0.0      0.0      0.00              0.00         0    0.000       0      0
 Sum    529/0   32.45 GB   0.0    104.1    44.1     60.0     118.3     58.3       5.4   7.7    169.8    192.9    628.00            479.19       840    0.748    622M    17M
 Int      0/0    0.00 KB   0.0      0.0     0.0      0.0       0.0      0.0       0.0   0.0      0.0      0.0      0.00              0.00         0    0.000       0      0

** Compaction Stats [default] **
Priority    Files   Size     Score Read(GB)  Rn(GB) Rnp1(GB) Write(GB) Wnew(GB) Moved(GB) W-Amp Rd(MB/s) Wr(MB/s) Comp(sec) CompMergeCPU(sec) Comp(cnt) Avg(sec) KeyIn KeyDrop
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Low      0/0    0.00 KB   0.0    104.1    44.1     60.0     103.0     43.0       0.0   0.0    187.8    185.7    567.98            441.10       395    1.438    622M    17M
High      0/0    0.00 KB   0.0      0.0     0.0      0.0      15.3     15.3       0.0   0.0      0.0    260.7     59.93             38.09       444    0.135       0      0
User      0/0    0.00 KB   0.0      0.0     0.0      0.0       0.0      0.0       0.0   0.0      0.0    266.8      0.09              0.00         1    0.087       0      0
Uptime(secs): 553803.8 total, 0.0 interval
Flush(GB): cumulative 15.284, interval 0.000
AddFile(GB): cumulative 0.000, interval 0.000
AddFile(Total Files): cumulative 0, interval 0
AddFile(L0 Files): cumulative 0, interval 0
AddFile(Keys): cumulative 0, interval 0
Cumulative compaction: 118.31 GB write, 0.22 MB/s write, 104.14 GB read, 0.19 MB/s read, 628.0 seconds
Interval compaction: 0.00 GB write, 0.00 MB/s write, 0.00 GB read, 0.00 MB/s read, 0.0 seconds
Stalls(count): 0 level0_slowdown, 0 level0_slowdown_with_compaction, 0 level0_numfiles, 0 level0_numfiles_with_compaction, 0 stop for pending_compaction_bytes, 0 slowdown for pending_compaction_bytes, 0 memtable_compaction, 0 memtable_slowdown, interval 0 total count

**Comment by xiaofan-luan** [comment]
The data increase is a expect behavior, even if user didn't ingest any data, the system will ingest time tick into it every 100ms.
Rmq will purge expired data after 7 days, you may want to reduce the time by change rocksmq.retentionTimeInMinutes.

However, the 3-4GB memory leakage is weird, I will investigating on that

**Comment by rchangj** [comment]
@xiaofan-luan Is it ok to remove old .sst files? Is there any configuration that allows automatic removal of old files?

**Comment by rchangj** [comment]
@xiaofan-luan From what's is under /volume, it seems all the .sst files are kept. I didn't see files older than 7 days have been purged. Feel free to let me know if I miss anything. Thanks for the help.

**Comment by fishpenguin** [comment]
@rchangj There is an autimatic removal policy of old files, the expired time is 7 days, and the unit of remove is page, one page is 2GB currently,  it means that if the message size is smaller than 2G, the file won't be expired, and this policy is processed by milvus standalone, so if you want to do removal automatically, you need to keep milvus standalone started after 7days.
So the .sst files olderd than 7days may not be deleted.

**Comment by fishpenguin** [comment]
@rchangj  We are still working on the memory leak issue, and this will be fixed in 2.0.0-GA.

**Comment by rchangj** [comment]
Hi, @fishpenguin Just FYI, I'm running the latest image following recommendation from: 
milvusdb/milvus-dev:master-20211214-dbaca0a
https://github.com/milvus-io/milvus/issues/12497
I'm still seeing memory leak.

**Comment by fishpenguin** [comment]
@rchangj The memory leak is because that the block cache usage of rocksdb in rocksmq is not limited, milvus keeps sending timetick messages to rocksmq when idle and the rocksdb block cache size keeps growing. We are trying to figure out why the block cache size is unlimited and this will be fixed soon.

BTW, did you do some operations in standalone before kept the standalone idle? Or just did nothing since milvus stabdalone started.

**Comment by rchangj** [comment]
@fishpenguin I didn't do anything this time.

**Comment by xiaofan-luan** [comment]
if we keep writing data into milvus we should see memory going up a little bit, which should be fine.
The memory going up is from bloomfilters and index blocks, which are related to datasize. Block cache, on the other side, is stick to 1GB as expexted.
trying to reproduce on a standalone cluster but failed, probably need a way to reproduce

**Comment by rchangj** [comment]
In my setup, I have let the standalone nodes running idle for 3 weeks. I saw about 60G memory reduced from free memory. They seems go to buff/cache.

**Comment by yanliang567** [comment]
@rchangj we had a fix for this memory leak on PreGA release, could you please help to retry on it?

**Comment by xiaofan-luan** [comment]
> buff/cache

if it went to buff/cache I doubt this is a memory leakage. probably just because the system is using some memory as page cache and the memory will return to other process once system memory usage is too high

you can check rss of the milvus see if rss increase，by default settings without any data milvus should use no more than few giga bytes memory

**Comment by rchangj** [comment]
@yanliang567 I currently use this image: master-20211214-dbaca0a which seems has the fix for https://github.com/milvus-io/milvus/issues/12497.   Does it have the fix for the memory issue?

**Comment by yanliang567** [comment]
@rchangj I suggest you reinstall a milvus with PreGA, as there are a few fixes for standalone memory, such as #13865, #13878, #13683,etc.

**Comment by xiaofan-luan** [comment]
> @yanliang567 Yes, I have already installed preGA. The memory leak issue seems much better but there is still some loss. For the last two days, I'm seeing ~500k/day which is much lower than 4G/day.

Cool, that is probably because we are still writing timetick into the system and rocksdb need more memory for index and bloomfilter.

Once we reach 7 days and we begin to purge data in rocksmq then data will not keep increasing~~
